tests/baselines/reference/user/lodash.log

```diff
@@ -1,427 +1,21 @@
-  Types of property 'maxWait' are incompatible.
-    Type 'number | undefined' is not assignable to type 'number'.
-      Type 'undefined' is not assignable to type 'number'.
-node_modules/lodash/toLower.js(11,21): error TS8024: JSDoc '@param' tag has name 'string', but there is no parameter with that name.
-node_modules/lodash/toUpper.js(11,21): error TS8024: JSDoc '@param' tag has name 'string', but there is no parameter with that name.
-node_modules/lodash/transform.js(46,37): error TS2554: Expected 0-1 arguments, but got 2.
-node_modules/lodash/transform.js(60,12): error TS2722: Cannot invoke an object which is possibly 'undefined'.
-node_modules/lodash/truncate.js(60,36): error TS2532: Object is possibly 'undefined'.
-node_modules/lodash/truncate.js(61,26): error TS2532: Object is possibly 'undefined'.
-node_modules/lodash/truncate.js(62,30): error TS2532: Object is possibly 'undefined'.
-node_modules/lodash/truncate.js(78,16): error TS2454: Variable 'strSymbols' is used before being assigned.
-node_modules/lodash/truncate.js(85,7): error TS2454: Variable 'strSymbols' is used before being assigned.
-node_modules/lodash/unary.js(19,10): error TS2554: Expected 3 arguments, but got 2.
-node_modules/lodash/unescape.js(30,37): error TS2769: No overload matches this call.
-  The last overload gave the following error.
-    Argument of type 'Function' is not assignable to parameter of type '(substring: string, ...args: any[]) => string'.
-node_modules/lodash/union.js(23,42): error TS2345: Argument of type '(value: any) => boolean' is not assignable to parameter of type 'boolean | undefined'.
-  Type '(value: any) => boolean' is not assignable to type 'true'.
-node_modules/lodash/unionBy.js(36,42): error TS2345: Argument of type '(value: any) => boolean' is not assignable to parameter of type 'boolean | undefined'.
-  Type '(value: any) => boolean' is not assignable to type 'true'.
-node_modules/lodash/unionBy.js(36,91): error TS2554: Expected 0-1 arguments, but got 2.
-node_modules/lodash/unionWith.js(31,42): error TS2345: Argument of type '(value: any) => boolean' is not assignable to parameter of type 'boolean | undefined'.
-  Type '(value: any) => boolean' is not assignable to type 'true'.
-node_modules/lodash/uniqBy.js(28,75): error TS2554: Expected 0-1 arguments, but got 2.
-node_modules/lodash/wrapperAt.js(34,17): error TS2339: Property 'slice' does not exist on type 'LazyWrapper'.
-node_modules/lodash/wrapperAt.js(40,51): error TS2339: Property 'thru' does not exist on type 'LodashWrapper'.
-node_modules/lodash/wrapperReverse.js(33,23): error TS2339: Property 'reverse' does not exist on type 'LazyWrapper'.
-node_modules/lodash/xorBy.js(36,81): error TS2554: Expected 0-1 arguments, but got 2.
 
 
 
 Standard error:
+
+../../../src/compiler/utilities.ts:4805
+        return find(symbol.declarations, isClassLike);
+                           ^
+TypeError: Cannot read property 'declarations' of undefined
```

Comment:
crashes in core.js:we don't expect to see `@private` on a constructor function like LodashWrapper. 

```js
/** @private */
function LodashWrapper(value) {
  this.__wrapped__ = value
}
```

Previously we didn't retrieve any modifiers, but now we see `@private` and try to create a type structure equivalent to

```ts
declare class LodashWrapper {
  private constructor(value: any);
}
```

But

1. This is not what is intended; the lodash authors intend to say that LodashWrapper is not *exported*.
2. The code assumes that it's running on a real constructor not a constructor function, so it assumes `declaration.parent.symbol` is the symbol of the containing class, when it should just be `declaration.symbol`.

I'll try to create a small repro and then decide what the correct behaviour is in the bug.

Reply:
#35766